packages/jest-jasmine2/src/index.ts

```diff
@@ -106,7 +108,7 @@ export default async function jasmine2(
     if (config.resetMocks) {
       runtime.resetAllMocks();
 
-      if (config.timers === 'legacy') {
+      if (config.fakeTimers.legacyFakeTimers) {
```

Comment:
```suggestion
      if (config.fakeTimers.enableGlobally && config.fakeTimers.legacyFakeTimers) {
```

Reply:
Just checked. If code is changed as suggested, it does not work in case if user has just `{legacyFakeTimers: true}` in Jest config and calls `jest.useFakeTimers()` in top-level in a test file (i.e. installs legacy timers just for that file). See fixed `e2e/fake-timers-legacy/reset-mocks` test case.

Reply:
but if config is `{enableGlobally: false, legacyFakeTimers:true}`, we should not call `useFakeTimers()`

Reply:
Right.. I will revert the test and will fix this in Circus as well. It has similar logic.